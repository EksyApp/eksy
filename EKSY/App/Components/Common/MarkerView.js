import React, {Component} from 'react'
import {View, StyleSheet} from "react-native";
import Pointer from "./Pointer";
import Label from "./Label";
import TextArea from "./TextArea";
import PictureSwiper from "./PictureSwiper";
import Message from "./Message";
import PropTypes from 'prop-types';
import {MarkerShape, RegionShape} from "../../Utils/PropTypeShapes";

export default class MarkerView extends Component {
	
	getStatusMessage() {
		switch(this.props.marker.status) {
			case -1:
				return "Rejected"
			case 0:
				return "Waiting for confirmation"
			case 1:
				return "Accepted"
		}
	}
	
	render() {
		return(
				<View>
					<Pointer
							initialRegion={{...this.props.currentRegion, latitude: this.props.marker.latitude, longitude: this.props.marker.longitude}}
							style={styles.map}
					/>
					<Label>{this.props.marker.title}</Label>
					<Message>{this.getStatusMessage()}</Message>
					<TextArea>{this.props.marker.text}</TextArea>
					{this.props.marker.images && <PictureSwiper style={styles.swiper} data={this.props.marker.images}/>}
					{this.props.marker.filters && this.props.marker.filters.map((filter) => <Message>{filter}</Message>)}
				</View>
		)
	}
	
}

const styles = StyleSheet.create({
	map: {
		width: '100%',
		height: 300,
		marginBottom: 30
	},
	
	swiper: {
		width: '100%',
		height: 300,
		marginTop: 30
	}
})

MarkerView.propTypes = {
	marker: MarkerShape,
	currentRegion: RegionShape
}