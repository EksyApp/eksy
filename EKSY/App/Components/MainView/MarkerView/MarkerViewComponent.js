import React, {Component} from 'react'
import {Modal, ScrollView, TouchableWithoutFeedback, View, Dimensions, StyleSheet} from 'react-native'
import Label from "../../Common/Label";
import ViewMoreText from "../../Common/ViewMoreText";
import * as Theme from "../../../Theme/Colors";
import Divider from "../../Common/Divider";
import Picture from "../../Common/Picture";
import {Actions} from 'react-native-router-flux'
import PictureSwiper from "../../Common/PictureSwiper";
import {Icon} from 'react-native-elements'
import TextArea from "../../Common/TextArea";


const Screen = {
	width: Dimensions.get('window').width,
	height: Dimensions.get('window').height - 75
}

class MarkerViewComponent extends Component {
	
	
	constructor(props) {
		super(props)
		this.itemWidth = (0.85 * Screen.width * 2) / 3
		this.sliderWidth = 0.85 * Screen.width
		this.state = {
			isCollapsed: true
		}
	}
	
	_setCollapsed = () => {
		if (!this.state.isCollapsed) {
			this.setState({isCollapsed: true})
		}
	}
	
	_setExpanded = () => {
		if (this.state.isCollapsed) {
			this.setState({isCollapsed: false})
		}
	}
	
	
	_renderImages() {
		if (this.props.marker.images && this.props.marker.images.length > 0) {
			return (
					<View
							style={[!this.state.isCollapsed && styles.renderImagesCollapsed, this.state.isCollapsed && styles.renderImagesExpanded]}>
						<View style={styles.labelWrapper}>
							<Divider/>
						</View>
						
						<PictureSwiper data={this.props.marker.images}/>
					</View>
			)
		} else {
			return null
		}
	}
	
	_renderText() {
		if (this.props.marker.images && this.props.marker.images.length > 0) {
			return (<ViewMoreText
							numberOfLines={1}
							afterCollapse={this._setCollapsed}
							afterExpand={this._setExpanded}
					>
						{this.props.marker.text}
					</ViewMoreText>
			)
		} else {
			return (
					<TextArea>
								{this.props.marker.text}
					</TextArea>
			)
		}
		
	}
	
	_renderEditingIcon() {
		if (this.props.user && this.props.marker && this.props.marker.creationInfo && this.props.user.uid === this.props.marker.creationInfo.user) {
			return(
					<Icon
							name="edit"
							size={25}
							containerStyle={styles.editIcon}
							onPress={() => {
								Actions.editMarker()
								this.props.setMarkerViewHidden()
							}}
					/>
			)
		}
	}
	
	_renderTitleAndText() {
		return (
				<View
						style={this.state.isCollapsed ? styles.titleAndTextWrapperCollapsed : styles.titleAndTextWrapperExpanded}
				>
					<View style={styles.titleContainer}>
						<Label style={styles.title}>{this.props.marker.title}</Label>
						{this._renderEditingIcon()}
					</View>
					
					<ScrollView>
						{this._renderText()}
					</ScrollView>
				</View>
		)
	}
	
	render() {
		return (
				<TouchableWithoutFeedback onPress={this.props.setMarkerViewHidden}>
					<Modal
							visible={this.props.markerViewVisible}
							animationType={'fade'}
							transparent
							onRequestClose={this.props.setMarkerViewHidden}
					>
						<View style={styles.content}>
							{this._renderTitleAndText()}
							{this._renderImages()}
						</View>
					</Modal>
				</TouchableWithoutFeedback>
		)
	}
}


const styles = StyleSheet.create({
	
	content: {
		backgroundColor: Theme.backgroundColor,
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center',
		borderRadius: 20,
		margin: 20
	},
	
	renderImagesCollapsed: {
		flex: 1,
		justifyContent: 'center'
	},
	renderImagesExpanded: {
		flex: 5,
		justifyContent: 'center',
		margin: 5
	},
	
	titleAndTextWrapperCollapsed: {
		flex: 1,
		alignItems: 'center',
		margin: 10
	},
	
	titleAndTextWrapperExpanded: {
		// flex: 0,
		alignItems: 'center',
		margin: 10,
	},
	
	swiper: {
		// width: 300,
		// margin: 20,
		width: '100%',
		height: '100%'
		// justifyContent: 'center',
		// alignItems: 'center',
	},
	
	title: {
		fontSize: 22,
		fontWeight: 'bold',
		color: Theme.accentColor
	},
	
	labelWrapper: {
		padding: 10
	},
	
	listStyle: {
		margin: 10
	},
	
	closeButton: {
		position: 'absolute',
		left: Screen.width - 120
	},
	
	editIcon: {
		position: 'absolute',
		top: 0,
		right: 0,
		
	},
	
	titleContainer: {
		width: '100%'
	}
})

export default MarkerViewComponent