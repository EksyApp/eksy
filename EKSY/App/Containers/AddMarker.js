import React, {Component} from 'react'
import {View, Text, TextInput, StyleSheet} from 'react-native'
// import {Container, Content, Header, Body, Title, Left, Button} from 'native-base'
import PointSelector from '../Components/PointSelector'
import Marker from '../Map/Marker'
import MapManager from '../Map/MapManager'
import Header from '../Components/Header'
import * as ReduxActions from '../Actions'
import {Actions} from 'react-native-router-flux'
import {connect} from 'react-redux'
import {backgroundColor, frontgroundColor} from '../Theme'
import {Grid, Row, FormInput, Button} from 'react-native-elements'
import ActionButton from '../Components/ActionButton'

class AddMarker extends Component {
	
	constructor(props) {
		super(props)
		
		this.state = {
			text: ""
		}
		
		this.mapManager = new MapManager();
	}
	
	addNewMarker() {
		let latitude = this.props.currentRegion.latitude;
		let longitude = this.props.currentRegion.longitude;
		let marker = <Marker latitude={latitude} longitude={longitude} text={this.state.text}/>;
		this.mapManager.addMarker(marker);
		setTimeout(() => this.mapManager.flyToPosition(latitude, longitude), 1000)
		Actions.mapView();
		// this.mapManager.update();
	}
	
	
	render() {
		return (
				// <Container>
				//   <Header>
				//
				//     <MenuButton />
				//     <Left />
				//     <Body>
				//       <Title>Add a new marker</Title>
				//     </Body>
				//   </Header>
				//   <Content style={Style.container}>
				//     <PointSelector onChange={(region) => this.state.region = region} style={Style.map} />
				//     <TextInput
				//       onChange={(event) => this.state.text = event.nativeEvent.text}
				//       multiline
				//       numberOfLines={10}
				//       style = {Style.textInput}
				//     />
				//     <Button block onPress={() => this.addNewMarker()} style={Style.button}>
				//       <Text style={Style.text}>Add</Text>
				//     </Button>
				//   </Content>
				//
				// </Container>
				
				<View style={styles.container}>
					<Header title="Add Marker" menuButtonPress={this.props.menuButtonPress}/>
					<Grid>
						<Row size={2} containerStyle={styles.row}>
							<PointSelector onChange={(region) => {
								this.props.regionChange(region)
							}} currentRegion={this.props.currentRegion} style={styles.map}/>
						</Row>
						<Row size={2} containerStyle={styles.row}>
							<FormInput onChangeText={(text) => this.setState({text: text})} multiline numberOfLines={8} style={styles.textArea}/>
						</Row>
						<Row size={1} containerStyle={styles.row}>
							<ActionButton title="Add" onPress={() => {
								this.addNewMarker()
							}}/>
						</Row>
					</Grid>
				</View>
		)
	}
}

const styles = StyleSheet.create({
	container: {
		backgroundColor: backgroundColor,
		flex: 1
	},
	
	row: {
		width: '100%'
	},
	
	map: {
		flex: 1,
	},
	
	textArea: {
		flex: 1,
		textAlignVertical: 'top',
		width: '100%',
		backgroundColor: frontgroundColor,
		marginTop: 10
	}
	
	
})

const mapStateToProps = (state) => {
	return {
		currentRegion: state.map.currentRegion
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		menuButtonPress: () => {
			dispatch(ReduxActions.drawerOpen())
		},
		regionChange: (region) => {
			dispatch(ReduxActions.updateRegion(region))
		}
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(AddMarker)
