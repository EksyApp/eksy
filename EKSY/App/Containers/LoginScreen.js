
import {
  Text,
  View,
  StyleSheet,
  dismissKeyboard,
  TouchableWithoutFeedback
} from 'react-native'

import {
    Grid,
    Row
} from 'react-native-elements'

import Header from '../Components/Header'
import MenuButton from '../Components/MenuButton'
import Button from '../Components/Button'
import Input from '../Components/Input'
import React, {Component} from 'react'
import * as firebase from 'firebase'
import DismissKeyboard from 'dismissKeyboard'
import PostOffice from '../lib/PostOffice'
import {connect} from 'react-redux'
import * as Actions from '../Actions'
import * as Theme from '../Theme'

class LoginScreen extends Component {
  constructor (props) {
    super(props)

    this.state = {
      email: '',
      password: '',
      response: ''
    }

    this.po = new PostOffice()
  }

  async signup () {
    DismissKeyboard()

    try {
      await firebase.auth().createUserWithEmailAndPassword(this.state.email, this.state.password)

      this.setState({
        response: 'account created'
      })

      this.po.getPacket('user').email = this.state.email
    } catch (error) {
      this.setState({
        response: error.toString()
      })
    }
  }

  async login () {
    DismissKeyboard()

    try {
      await firebase.auth().signInWithEmailAndPassword(this.state.email, this.state.password)

      this.setState({
        response: 'Logged In!'
      })

      this.po.getPacket('user').email = this.state.email
    } catch (error) {
      this.setState({
        response: error.toString()
      })
    }
  }

  render () {
    return (
      <View style={styles.container}>
        <Header title='Login/Signup' menuButtonPress={this.props.menuButtonPress} />
        <View>
          <Input
            label='Email'
            value={this.state.email}
            onChangeText={email => this.setState({ email })}
            placeholder='user@gmail.com'
          />
        </View>
        <View>
          <Input
            label='Password'
            value={this.state.password}
            onChangeText={password => this.setState({ password })}
            placeholder='your password'
            secureTextEntry
        />
        </View>
        <View style={styles.buttons}>
          <Button onPress={() => this.login()}>
          Log in
          </Button>
          <Button onPress={() => this.signup()}>
          Sign up
          </Button>
        </View>
        <View style={styles.response}>
          <Text style={styles.responseText}>{this.state.response}</Text>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: Theme.backgroundColor
  },

  buttons: {
    marginTop: 50
  },

  responseText: {
    textAlign: 'center',
    width: '100%',
    fontSize: 18,
    color: 'white'
  },

  response: {
    marginTop: 50
  }

})

const mapStateToProps = (state) => {
  return {}
}

const mapDispatchToProps = (dispatch) => {
  return {
    menuButtonPress: () => { dispatch(Actions.drawerOpen()) }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(LoginScreen)
