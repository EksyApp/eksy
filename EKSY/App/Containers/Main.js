import React, {Component} from 'react'
import NavigationRouter from '../Navigation/NavigationRouter'
import FirebaseConfig from '../includes/FirebaseConfig'
import {Provider} from 'react-redux'
import {ActivityIndicator} from 'react-native'
import Store from '../Store'

class Main extends Component {

  state = {
    store: null
  }

  constructor (props) {
    super(props)
    FirebaseConfig.initialise()
  }

  async componentWillMount() {
    const store = await Store()
    this.setState({ store })
  }

  render () {
    if (!this.state.store) {
      return (
        <ActivityIndicator />
      )
    }

    return (
      <Provider store={this.state.store}>
        <NavigationRouter />
      </Provider>
    )
  }
}

export default Main
