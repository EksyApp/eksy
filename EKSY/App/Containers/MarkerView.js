import React, {Component} from 'react'
import {View, ScrollView, Text, StyleSheet} from 'react-native'
import Header from "../Components/Header";
import PictureList from '../Components/PictureList'
import * as ReactActions from '../Actions'
import {connect} from 'react-redux'
import TextArea from '../Components/TextArea'
import * as Theme from '../Theme'
import Divider from '../Components/Divider'
import Label from '../Components/Label'

export class MarkerView extends Component {

	_renderImages() {
		if(this.props.marker.images.length > 0) {
			return(
					<View>
						<Divider />
						<Label>Images</Label>
						<PictureList data={this.props.marker.images} listStyle={style.listStyle} imageContainerStyle={style.imageContainer} />
					</View>
			)
		} else {
			return null
		}
	}

	render() {
		return(
				<View style={style.container}>
					<Header backButton title={this.props.marker.title}/>
					<ScrollView>
						<View>
							<TextArea>
								{this.props.marker.text}
							</TextArea>
						</View>

						{this._renderImages()}

					</ScrollView>
				</View>
		)
	}

}

const style = StyleSheet.create({
	imageContainer: {
		width: '100%',
	},

	listStyle: {
		margin: 10
	},

	container: {
		backgroundColor: Theme.backgroundColor,
		flex: 1
	},

})

const mapStateToProps = (state) => {
	return {
		marker: state.map.selectedMarker.marker
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(MarkerView)
