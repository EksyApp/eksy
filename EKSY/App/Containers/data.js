export const characters = [
  {
    name: 'Kevin',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane03/sub-buzz-3484-1481210659-5.png',
    coordinate: [ 40.764466, -73.974488 ],
    good: true,
  },
  {
    name: 'Uncle Frank',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane02/sub-buzz-25788-1481210705-7.png',
    coordinate: [ 40.764656, -73.980907 ],
    good: false,
  },
  {
    name: 'Aunt Leslie',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane01/sub-buzz-5211-1481210889-4.png',
    coordinate: [ 40.764838, -73.981193 ],
    good: false,
  },
  {
    name: 'The guy with the same coat as Kevin’s dad',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane03/sub-buzz-4392-1481212051-3.jpg',
    coordinate: [ 40.748418, -73.985702 ],
    good: false,
  },
  {
    name: 'Donald Trump',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane03/sub-buzz-5109-1481213313-1.png',
    coordinate: [ 40.762389, -73.973817 ],
    good: false,
  },
  {
    name: 'The lady from Plaza Hotel reservations',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-6769-1481213418-3.png',
    coordinate: [ 40.779707, -73.979012 ],
    good: false,
  },
  {
    name: 'This scary ass big-tilted Santa',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane02/sub-buzz-27082-1481212691-2.png',
    coordinate: [ 40.747067, -74.004496 ],
    good: false,
  },
  {
    name: 'Random cousins/siblings',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-7903-1481215701-1.png',
    coordinate: [ 40.765298, -73.976022 ],
    good: false,
  },
  {
    name: '“Watch it kid HAHAHAHA” guy',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-8357-1481215914-1.png',
    coordinate: [ 40.791939, -73.963466 ],
    good: false,
  },
  {
    name: 'The taxi driver near Central Park',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-8061-1481216019-8.png',
    coordinate: [ 40.792364, -73.964200 ],
    good: false,
  },
  {
    name: 'The prostitutes near Central Park',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane03/sub-buzz-6355-1481215967-1.png',
    coordinate: [ 40.791646, -73.964681 ],
    good: false,
  },
  {
    name: 'Buzz',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane03/sub-buzz-4009-1481211103-1.png',
    coordinate: [ 40.764298, -73.971752 ],
    good: false,
  },
  {
    name: 'Esther Stone, hotel lady',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-6874-1481213534-3.png',
    coordinate: [ 40.764116, -73.973581 ],
    good: false,
  },
  {
    name: 'Fuller',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane02/sub-buzz-26273-1481211159-3.png',
    coordinate: [ 40.763241, -73.975782 ],
    good: false,
  },
  {
    name: 'Mrs. McCallister',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane02/sub-buzz-25796-1481211030-8.png',
    coordinate: [ 40.758702, -73.978598 ],
    good: false,
  },
  {
    name: 'Mr. McCallister',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/10/asset/buzzfeed-prod-fastlane03/sub-buzz-4020-1481211789-2.png',
    coordinate: [ 40.758889, -73.985125 ],
    good: false,
  },
  {
    name: 'The NYC cop who takes Mrs. McCallister to Rockefeller Plaza',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/13/asset/buzzfeed-prod-fastlane01/sub-buzz-21202-1481220551-1.png',
    coordinate: [ 40.771156, -73.979802 ],
    good: true,
  },
  {
    name: 'The woman that slaps/punches Marv twice',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane03/sub-buzz-5073-1481213214-2.png',
    coordinate: [ 40.761432, -73.977645 ],
    good: false,
  },
  {
    name: 'Harry',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-6492-1481212963-3.png',
    coordinate: [ 40.792084, -73.966925 ],
    good: false,
  },
  {
    name: 'Tim Curry',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane01/sub-buzz-6839-1481213501-1.png',
    coordinate: [ 40.764859, -73.975421 ],
    good: true,
  },
  {
    name: 'Duncan',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane02/sub-buzz-28268-1481214964-2.png',
    coordinate: [ 40.763560, -73.972198 ],
    good: true,
  },
  {
    name: 'Marv',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane02/sub-buzz-27059-1481212962-1.png',
    coordinate: [ 40.791871, -73.966812 ],
    good: false,
  },
  {
    name: 'The Pigeon Lady',
    image: 'https://img.buzzfeed.com/buzzfeed-static/static/2016-12/8/11/asset/buzzfeed-prod-fastlane02/sub-buzz-27109-1481213135-7.png',
    coordinate: [ 40.765126, -73.979924 ],
    good: true,
  },
];
