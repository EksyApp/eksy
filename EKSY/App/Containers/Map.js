import React, { Component } from 'react';
import {
  StyleSheet,         // CSS-like styles
  Text,               // Renders text
  TouchableOpacity,   // Pressable container
  View                // Container component
} from 'react-native';
import MapView from 'react-native-maps';
// Import data
import { characters } from './data';
import Callout from './Callout';
import styles from './Styles/MapStyles'

export default class Map extends Component {

  state = {
    // Show good or all characters flag
    showGoodOnly: false,
  }

  render() {
    return (
      <View style={styles.container}>
      {/* Map*/}
      <MapView
      style={styles.map}
      // Position on Manhattan, New York
      initialRegion={{
        latitude: 40.77096,
        longitude: -73.97702,
        latitudeDelta: 0.0491,
        longitudeDelta: 0.0375,
      }}
      >
      {/* Loop through characters and add pins on the map */}
      {characters.map((character, index) =>
            // If showGoodOnly is true, but the character is bad - do not show it
            this.state.showGoodOnly && !character.good || <MapView.Marker
              coordinate={{
                latitude: character.coordinate[0],
                longitude: character.coordinate[1],
              }}
              // Callout offset
              calloutOffset={{ x: -8, y: 28 }}
              // Greed color for good characters and red for others
              pinColor={character.good ? '#009688' : '#f44336'}
              key={index}
            >
              {/* Callout */}
              <MapView.Callout tooltip style={styles.callout}>
                <Callout
                  name={character.name}
                  image={character.image}
                />
              </MapView.Callout>
            </MapView.Marker>
          )}
      </MapView>
      {/* Button */}
      <View style={styles.buttonContainer}>
      <TouchableOpacity
      style={styles.button}
      // Toggle this.state.showGoodOnly
      onPress={() => this.setState({
        showGoodOnly: !this.state.showGoodOnly
      })}
      >
      <Text>{this.state.showGoodOnly ? 'Show All' : 'Show Good Only'}</Text>
      </TouchableOpacity>
      </View>
      </View>
    );
  }
}
