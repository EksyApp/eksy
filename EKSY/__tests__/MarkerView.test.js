import 'react-native';
import React from 'react';
import {MarkerView} from '../App/Containers/MarkerView';
import renderer from 'react-test-renderer';

test('MarkerView renders correctly', () => {
  const tree = renderer.create(
    <MarkerView marker={{title: 'test', text: 'moar test', images:[{width: 10, height: 10, uri:"test"}]}}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
