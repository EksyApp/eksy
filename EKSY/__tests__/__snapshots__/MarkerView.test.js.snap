// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkerView renders correctly 1`] = `
<Modal
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  animationType="fade"
  hardwareAccelerated={false}
  hitSlop={undefined}
  onLayout={undefined}
  onRequestClose={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={undefined}
  testID={undefined}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgb(243,244,229)",
        "borderRadius": 20,
        "flex": 1,
        "justifyContent": "center",
        "margin": 20,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "margin": 10,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "rgb(101,114,122)",
            "fontSize": 22,
            "fontWeight": "bold",
          }
        }
      >
        test
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "rgb(89,85,78)",
              "fontSize": 18,
              "margin": 10,
            },
            undefined,
          ]
        }
      >
        text
      </Text>
    </View>
  </View>
</Modal>
`;
